OrbitCore/ConnectionManager.cpp

```diff
@@ -210,33 +212,54 @@ void ConnectionManager::SetupServerCallbacks() {
         uint32_t pid =
             static_cast<uint32_t>(msg.m_Header.m_GenericHeader.m_Address);
 
-        PRINT("RemoteModuleDebugInfo pid=%d\n", pid);
         std::shared_ptr<Process> process = process_list_.GetProcess(pid);
         if (!process) {
-          PRINT("Process not found (pid=%d)\n", pid);
+          ERROR("Process not found (pid=%d)", pid);
           return;
         }
 
         Capture::SetTargetProcess(process);
 
-        std::vector<ModuleDebugInfo> remoteModuleDebugInfo;
-        std::vector<std::string> modules;
-
+        std::vector<ModuleDebugInfo> module_infos;
         std::istringstream buffer(std::string(msg.m_Data, msg.m_Size));
         cereal::BinaryInputArchive inputAr(buffer);
-        inputAr(modules);
-
-        for (std::string& module : modules) {
-          ModuleDebugInfo moduleDebugInfo;
-          moduleDebugInfo.m_Name = module;
-          process->FillModuleDebugInfo(moduleDebugInfo);
-          remoteModuleDebugInfo.push_back(moduleDebugInfo);
+        inputAr(module_infos);
+
+        std::vector<ModuleDebugInfo> module_infos_send_back;
+
+        for (auto& module_info : module_infos) {
+          std::shared_ptr<Module> module =
+              process->GetModuleFromName(module_info.m_Name);
+          if (!module) {
+            ERROR("Unable to find module %s", module_info.m_Name.c_str());
+            continue;
+          }
+
+          if (!module_info.m_Functions.empty()) {
+            LOG("Received %lu function symbols from local machine for "
+                "module %s",
+                module_info.m_Functions.size(), module_info.m_Name.c_str());
+            SymbolHelper::LoadSymbolsFromDebugInfo(module, module_info);
+            continue;
+          }
+
+          if (SymbolHelper::LoadSymbolsCollector(module)) {
+            SymbolHelper::FillDebugInfoFromModule(module, module_info);
+            LOG("Loaded %lu function symbols for module %s",
+                module_info.m_Functions.size(), module_info.m_Name.c_str());
+          } else {
+            ERROR("Unable to load symbols of module %s",
+                  module->m_Name.c_str());
+          }
+
+          module_infos_send_back.emplace_back(module_info);
         }
 
         // Send data back
-        std::string messageData = SerializeObjectBinary(remoteModuleDebugInfo);
-        GTcpServer->Send(Msg_RemoteModuleDebugInfo, (void*)messageData.data(),
-                         messageData.size());
+        std::string message_data =
+            SerializeObjectBinary(module_infos_send_back);
+        GTcpServer->Send(Msg_RemoteModuleDebugInfo, (void*)message_data.data(),
```

Comment:
nit: I think you can just send message_data here.

Reply:
done